src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -626,7 +687,10 @@ private void markDuplicatePairs(final List<ReadEndsForMarkDuplicates> list) {
         for (final ReadEndsForMarkDuplicates end : list) {
             if (end != best) {
                 addIndexAsDuplicate(end.read1IndexInFile);
-                addIndexAsDuplicate(end.read2IndexInFile);
+
+                //in query-sorted case, these will be the same.
+                // TODO: also in coordinate sorted, when one read is unmapped
```

Comment:
Yes...well, I wasn't sure if the change is appropriate for this PR. Do you think it matters? I can just as easily change the behavior for coordinate-sorted input...